### PR TITLE
JBPM-8367: Stunner - Add marshaller test cases for Adhoc sub-process

### PR DESCRIPTION
**Pull request by LuboTerifaj**
Marshaller test cases for Ad-hoc sub-process.

Old marshaller has a few bugs, therefore some tests are ignored.
For more information see: https://issues.jboss.org/browse/JBPM-8462

@wmedvede , @hasys can you have a look please?

Thanks!

### CODE REVIEW

**Comment by LuboTerifaj** [comment]
Jenkins retest please.

**Comment by hasys** [comment]
Hi @LuboTerifaj, in general looks good to me. But, some Ignored tests are passing for me locally and others in my opinion should be disabled only for the old marshaller. See my changes in https://github.com/hasys/kie-wb-common/tree/PR2616

**Comment by LuboTerifaj** [comment]
@hasys Which tests are passing for you locally?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/subprocesses/AdHocSubProcessTest.java [hunk 1]
```diff
@@ -16,175 +16,454 @@
...
-                     executionSet.getAdHocCompletionCondition().getValue().getScript());
-        assertEquals("mvel",
-                     executionSet.getAdHocCompletionCondition().getValue().getLanguage());
+    @Ignore("There is a bug in old marshaller. Once, the bug is fixed, the method should be removed only from this class, " +
+            "not from its superclass.\n" +
+            "For more information see https://issues.jboss.org/browse/JBPM-8462")
+    @Test
+    @Override
+    public void testMarshallTopLevelEmptyPropertiesSubProcess() {
+    }
```
[the remark this answers is not in the document]
Comment (hasys) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/subprocesses/AdHocSubProcessTest.java]: One more update, I tried to model the same ad-hoc sub-process and migrate it, and everything migrated fine, without any issues. @LuboTerifaj any chances you have malformed XML in your tests?

[the remark this answers is not in the document]
Comment (hasys) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/subprocesses/AdHocSubProcessTest.java]: It is not a migration issue, I found a reproducer.

I there are 2 Ad-hoc subprocesses with same variable name like "varName" after reopen, one of the sub-process will lost it's type. Reproducible also with Embedded sub-processes.

@tsurdilo, is it a new issue? I don't remember something like this, but not sure we tested it before as well.

[the remark this answers is not in the document]
Comment (hasys) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-backend/src/test/java/org/kie/workbench/common/stunner/bpmn/backend/service/diagram/marshalling/subprocesses/AdHocSubProcessTest.java]: I reported an issue for old designer to keep it track:
* [JBPM-8479](https://issues.jboss.org/browse/JBPM-8479) - Sub-process variable lose it's type

**Comment by hasys** [comment]
HI @LuboTerifaj, @wmedvede, @romartin,

Just for a record. I found a way how to reproduce this issue in Stunner based designer as well:
* [JBPM-8482](https://issues.jboss.org/browse/JBPM-8482) - Stunner - variables with same name can't have different types

**Comment by LuboTerifaj** [comment]
Hi @hasys , 

yes, I have removed migration tests for now, because there are some differences, so I want to add them in another PR. 

Anyway, there is an issue with sub-process variables, as already reported in:

- https://issues.jboss.org/browse/JBPM-8462 
- https://issues.jboss.org/browse/JBPM-8479
-- This issue is also related to Stunner right? Did you reported it also for Stunner?
- https://issues.jboss.org/browse/JBPM-8482

So, is there some action required from me except removing `Ignore` annotation and change methods with adding `if(isCurrentMarshallerIsOld()) { return }` statement to not test legacy marshaller?

Thanks.

**Comment by hasys** [comment]
Hi @LuboTerifaj,

Yes, I guess we should keep all tests running with new marshaller by disabling old marshaller. I updated my suggestion and added some minor fixes like typos. See https://github.com/hasys/kie-wb-common/commit/46e3d5891eb823432197ddf6a380146b5620f7ea (this commit can be just cherry-picked).

About issues:
* Most of the mentioned in [JBPM-8462](https://issues.jboss.org/browse/JBPM-8462) mostly inconsistencies and does not harm work at runtime. So we can keep this issue, but according to the plan to remove old marshaller entirely not sure how much sense to fix them.
* No Stunner is not affected by [JBPM-8479](https://issues.jboss.org/browse/JBPM-8479), but has similar one already reported in [JBPM-8482](https://issues.jboss.org/browse/JBPM-8482)
* See previous point.

So my suggestion is apply fixes I provided in https://github.com/hasys/kie-wb-common/commit/46e3d5891eb823432197ddf6a380146b5620f7ea and it is ready to go. Thank you!

**Comment by LuboTerifaj** [comment]
Hello @hasys ,

I didn't cherry-picked your commit because there was necessary to change a few things anyway.

The PR is ready to review again.

Thanks!

**Comment by hasys** [comment]
Hi @LuboTerifaj,

sorry, I didn't notice it before, but it looks like there are also nodes outside of sub-process as it was in [PR for Event sub-processes](https://github.com/kiegroup/kie-wb-common/pull/2627#issuecomment-492638064).

It looks like there is some consistent issue around it. Can I ask you to investigate the root cause of it? 

image

**Comment by LuboTerifaj** [comment]
Hi @hasys , 

I have already reported the issue: https://issues.jboss.org/browse/JBPM-8488

PR is ready to review again.

Thanks!

**Comment by romartin** [comment]
Jenkins execute full downstream build

**Pull request merged**